# I. Introduction

In this notebook DNA Methylation data is obtained manually from GDC by searching for the selected patient using their id (submitter_id). After obtaining the data for each patient, the data is then cleaned and downloaded to further explore in notebook 4.


# DNA Methylation Data Gathering Steps

This is mostly for records keeping. I am adding it in case you wanted to see how exactly I obtained the DNA Methylation data for every patient.

Steps followed to obtain DNA Methylation Data:

1. Go to gdc.cancer.gov (Its where all TCGA data is stored.)
2. Click on Lauch Data Portal.
3. Click Accept if asked.
4. Click on the green Projects button.
5. Under 'Primary Site' select 'Bronchus and Lung' and under 'Program' select 'TCGA'.
6. Click on the project name TCGA-LUAD.
7. Click on Explore Project Data button.
8. In the search bar on the left, enter the submitter_id of the patient. Do not hit enter.
9. From the dropdown results, select the patient.
10. Click on the case id.
11. Scroll to 'File Counts by Data Category', Find 'DNA Methylation' and click on the file count (usually a 1).
12. Click on the file name (.txt)
13. Click on Download.
14. Unzip, extract and save to intended folder.


# Group 2: Non - Smoking Group

In [1]:
import pandas as pd #Necessary package/s.

In [2]:
group2 = pd.read_csv("group2.csv") #import data

In [3]:
group2

Unnamed: 0  submitter_id  cigarettes_per_day  alcohol_intensity  \
0         232  TCGA-55-8508                 0.0                0.0   
1          54  TCGA-44-2661                 0.0                0.0   
2         320  TCGA-73-7499                 0.0                0.0   
3         401  TCGA-86-8672                 0.0                0.0   
4         137  TCGA-50-5930                 0.0                0.0   
5         202  TCGA-55-7816                 0.0                0.0   
6         241  TCGA-55-8619                 0.0                0.0   
7         417  TCGA-91-6848                 0.0                0.0   
8         336  TCGA-75-7030                 0.0                0.0   
9         293  TCGA-69-7760                 0.0                0.0   

   years_smoked alcohol_history  pack_years_smoked  gender  \
0           0.0    Not Reported                0.0  female   
1           0.0    Not Reported                0.0  female   
2           0.0    Not Reported                0.0  female   
3           0.0    Not Reported                0.0    male   
4           0.0    Not Reported                0.0    male   
5           0.0    Not Reported                0.0  female   
6           0.0    Not Reported                0.0  female   
7           0.0    Not Reported                0.0    male   
8           0.0    Not Reported                0.0    male   
9           0.0    Not Reported                0.0    male   

                        race vital_status               ethnicity  \
0  black or african american        Alive  not hispanic or latino   
1                      white        Alive  not hispanic or latino   
2                      white         Dead  not hispanic or latino   
3                      white         Dead  not hispanic or latino   
4                      white         Dead  not hispanic or latino   
5                      white         Dead  not hispanic or latino   
6                      white        Alive  not hispanic or latino   
7                      white        Alive  not hispanic or latino   
8               not reported        Alive            not reported   
9                      white        Alive  not hispanic or latino   

  vital_status.1 bcr_patient_barcode disease  
0          Alive        TCGA-55-8508    LUAD  
1          Alive        TCGA-44-2661    LUAD  
2           Dead        TCGA-73-7499    LUAD  
3           Dead        TCGA-86-8672    LUAD  
4           Dead        TCGA-50-5930    LUAD  
5           Dead        TCGA-55-7816    LUAD  
6          Alive        TCGA-55-8619    LUAD  
7          Alive        TCGA-91-6848    LUAD  
8          Alive        TCGA-75-7030    LUAD  
9          Alive        TCGA-69-7760    LUAD

For every submitter_id in the group 2 dataset above, I follow the data gathering steps. They will be labeled DMG2P1 :DNA Methylation Group 2 patient 1. Patient 1 is the first one on the list therefore, DMG2P1 = TCGA-55-8508's DNA Methylation data. 

In [4]:
#Loading in all 10 patients from group 2 (non-smokers group)
g2p1 = pd.read_csv('DMG2P1.txt',  delimiter = '\t', encoding = 'utf-8')
g2p2 = pd.read_csv('DMG2P2.txt',  delimiter = '\t', encoding = 'utf-8')
g2p3 = pd.read_csv('DMG2P3.txt',  delimiter = '\t', encoding = 'utf-8')
g2p4 = pd.read_csv('DMG2P4.txt',  delimiter = '\t', encoding = 'utf-8')
g2p5 = pd.read_csv('DMG2P5.txt',  delimiter = '\t', encoding = 'utf-8')
g2p6 = pd.read_csv('DMG2P6.txt',  delimiter = '\t', encoding = 'utf-8')
g2p7 = pd.read_csv('DMG2P7.txt',  delimiter = '\t', encoding = 'utf-8')
g2p8 = pd.read_csv('DMG2P8.txt',  delimiter = '\t', encoding = 'utf-8')
g2p9 = pd.read_csv('DMG2P9.txt',  delimiter = '\t', encoding = 'utf-8')
g2p10 = pd.read_csv('DMG2P10.txt',  delimiter = '\t', encoding = 'utf-8')

# DNA Methylation Data Cleaning
## Group 2 - Patient 1

In [5]:
g2p1.head()#This is what the DNA Methylation Data for this patient looks like.

Composite Element REF  Beta_value Chromosome      Start        End  \
0            cg00000029    0.289343      chr16   53434200   53434201   
1            cg00000108         NaN       chr3   37417715   37417716   
2            cg00000109         NaN       chr3  172198247  172198248   
3            cg00000165    0.262197       chr1   90729117   90729118   
4            cg00000236    0.812817       chr8   42405776   42405777   

                                         Gene_Symbol  \
0                                     RBL2;RBL2;RBL2   
1  C3orf35;C3orf35;C3orf35;C3orf35;C3orf35;C3orf3...   
2          FNDC3B;FNDC3B;FNDC3B;FNDC3B;FNDC3B;FNDC3B   
3                                                  .   
4                                              VDAC3   

                                           Gene_Type  \
0       protein_coding;protein_coding;protein_coding   
1  lincRNA;lincRNA;lincRNA;lincRNA;lincRNA;lincRN...   
2  protein_coding;protein_coding;protein_coding;p...   
3                                                  .   
4                                     protein_coding   

                                       Transcript_ID  \
0  ENST00000262133.9;ENST00000544405.5;ENST000005...   
1  ENST00000328376.8;ENST00000332506.6;ENST000004...   
2  ENST00000336824.7;ENST00000415807.5;ENST000004...   
3                                                  .   
4                                  ENST00000022615.7   

                                Position_to_TSS                CGI_Coordinate  \
0                                -221;-1420;222   CGI:chr16:53434489-53435297   
1  18552;18552;6505;31445;18143;447;18552;18552    CGI:chr3:37451927-37453047   
2       157692;158618;151333;71272;158587;71273  CGI:chr3:172039703-172040934   
3                                             .    CGI:chr1:90724932-90727247   
4                                         13872    CGI:chr8:42410918-42411241   

  Feature_Type  
0      N_Shore  
1            .  
2            .  
3      S_Shore  
4            .

In [6]:
g2p1.isna().sum() #Lets see how much of the data is missing and for which column.

Composite Element REF        0
Beta_value               89776
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

There seems to be quite a few missing values in the Beta_value column. Lets find out what percentage of the data is missing.

In [7]:
#Total count for Beta_value. Should be the same for all datasets.
len(g2p1['Beta_value']) 

485577

In [8]:
print((89776 / 485577) * 100 , '%') #Percent of missing Beta_value data.

18.4885198434028 %


There may be a better way of handling missing values for this dataset but for now I will be removing the rows for which Beta_value is not available. Another way to handle these values would be to set them to 0, but that would not be a good idea with this type of data because it would give those rows a meaning they truly dont have. This is due to the Beta_value ranging from 0 to 1.

In [9]:
g2p1 = g2p1.dropna() #Removing rows of missing values.

g2p1.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

In [10]:
#This is to make sure the values removed are the missing ones and nothing else.

len(g2p1['Beta_value']) 


395801

In [11]:
485577 - 89776 #total - missing values = properly cleaned dataset total

395801

Lets do the same data cleaning as above to every patient on group 2. 

## Group 2 - Patient 2

In [12]:
g2p2.head() #Lets see the data.

Composite Element REF  Beta_value Chromosome      Start        End  \
0            cg00000292    0.673595      chr16   28878779   28878780   
1            cg00002426    0.591519       chr3   57757816   57757817   
2            cg00003994    0.053257       chr7   15686237   15686238   
3            cg00005847    0.416658       chr2  176164345  176164346   
4            cg00006414         NaN       chr7  149125745  149125746   

                                  Gene_Symbol  \
0          ATP2A1;ATP2A1;ATP2A1;ATP2A1;ATP2A1   
1         SLMAP;SLMAP;SLMAP;SLMAP;SLMAP;SLMAP   
2                                       MEOX2   
3  AC009336.19;HOXD3;HOXD3;HOXD3;RP11-387A1.5   
4              RN7SL521P;ZNF398;ZNF425;ZNF425   

                                           Gene_Type  \
0  protein_coding;protein_coding;protein_coding;p...   
1  protein_coding;protein_coding;protein_coding;p...   
2                                     protein_coding   
3  protein_coding;protein_coding;protein_coding;p...   
4  misc_RNA;protein_coding;protein_coding;protein...   

                                       Transcript_ID  \
0  ENST00000357084.6;ENST00000395503.7;ENST000005...   
1  ENST00000295951.6;ENST00000295952.6;ENST000003...   
2                                  ENST00000262041.5   
3  ENST00000468418.4;ENST00000249440.4;ENST000004...   
4  ENST00000488398.3;ENST00000426851.5;ENST000003...   

             Position_to_TSS                CGI_Coordinate Feature_Type  
0     373;290;-1275;-465;-83   CGI:chr16:28879633-28880547      N_Shore  
1   1585;368;261;257;257;514    CGI:chr3:57756198-57757263      S_Shore  
2                        576    CGI:chr7:16399497-16399700            .  
3  13259;267;3453;27387;1372  CGI:chr2:176164685-176165509      N_Shore  
4           242;-672;602;562  CGI:chr7:149126122-149127136      N_Shore

In [13]:
g2p2.isna().sum() #Total sum of all missing values for every column.

Composite Element REF       0
Beta_value               2599
Chromosome                  0
Start                       0
End                         0
Gene_Symbol                 0
Gene_Type                   0
Transcript_ID               0
Position_to_TSS             0
CGI_Coordinate              0
Feature_Type                0
dtype: int64

In [14]:
len(g2p2["Beta_value"]) #Total count.

27578

The DNA Methylation dataset for this patient is much smaller (27578 vs 485577) than others from the same group. Thus I will select another patient for whom DNA methylation file is closer to the other patients. This is so that any comparisons made are fair.


## Choosing a New Patient 2 for Group 2

In [15]:
#All clinical data is imported to choose new pt for group 2.

clinicaldata = pd.read_csv("TCGA-LUAD_Clinical.csv") 

#There are too many columns. 
#Lets remove some and leave in some to get a good overview of the data.

clindata_copy = clinicaldata[['submitter_id', 'cigarettes_per_day', 
                              'alcohol_intensity', 'years_smoked',
                              'alcohol_history', 'pack_years_smoked', 
                              'gender', 'race', 'vital_status', 'ethnicity', 
                              'bcr_patient_barcode','disease']].copy()

clindata_copy = clindata_copy.fillna(0) #Replaces all NaNs with 0.

#This selects all data for which the exposure data is equal to zero.
nonsmokers = clindata_copy.loc[(clindata_copy['cigarettes_per_day'] == 0) 
                               & (clindata_copy['years_smoked'] == 0) 
                               & (clindata_copy['pack_years_smoked'] == 0)]

#Randomly selects 1 patient that meets the above requirements.
NewPatient2 = nonsmokers.sample(n = 1) 

#Saves the new patient info into a new file. For safekeeping.
NewPatient2.to_csv('NewPatient2.csv') 



In [16]:
np2 = pd.read_csv('NewPatient2.csv') #New Patient 2 info.

np2.head()#Lets see the new patient's submitter Id to get the DNA Methylation data.

Unnamed: 0  submitter_id  cigarettes_per_day  alcohol_intensity  \
0         309  TCGA-71-8520                 0.0                0.0   

   years_smoked alcohol_history  pack_years_smoked  gender   race  \
0           0.0    Not Reported                0.0  female  asian   

  vital_status               ethnicity bcr_patient_barcode disease  
0         Dead  not hispanic or latino        TCGA-71-8520    LUAD

This new patient is not present in the list of patients for group 2 and thus can represent patient 2.

In [17]:
#importing new patient 2 DNA Methylation data.
g2np2 = pd.read_csv('DMG2NP2.txt', delimiter = '\t', encoding = 'utf-8') 

In [18]:
g2np2.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89844
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [19]:
len(g2np2) 

485577

The size of the dataset is a lot better than the first patient 2, and can be used to analyze alongside the others in the group. 

In [20]:
g2np2 = g2np2.dropna() #Removing rows of missing values.

g2np2.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

## Group 2 - Patient 3

In [21]:
g2p3.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89791
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [22]:
len(g2p3) #Size of dataset.

485577

In [23]:
g2p3 = g2p3.dropna() #Removing rows of missing values.

g2p3.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

There seems to be a trend amongst the mising values. Perhaps all datasets have approximately the same value missing from the same column? If so, we can skip a few steps and go straight to removal of missing values.

## Group 2 - Patient 4

In [24]:
g2p4.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89562
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [25]:
len(g2p4) #Size of dataset.

485577

In [26]:
g2p4 = g2p4.dropna() #Removing rows of missing values.

g2p4.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

## Group 2 - Patient 5

In [27]:
g2p5.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89634
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [28]:
len(g2p5) #Size of dataset.

485577

In [29]:
g2p5 = g2p5.dropna() #Removing rows of missing values.

g2p5.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

## Group 2 - Patient 6

In [30]:
g2p6.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89847
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [31]:
len(g2p6) #Size of dataset.

485577

In [32]:
g2p6 = g2p6.dropna() #Removing rows of missing values.

g2p6.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

## Group 2 - Patient 7

In [33]:
g2p7.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89756
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [34]:
len(g2p7) #Size of dataset.

485577

In [35]:
g2p7 = g2p7.dropna() #Removing rows of missing values.

g2p7.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

## Group 2 - Patient 8

In [36]:
g2p8.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89996
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [37]:
len(g2p8) #Size of dataset.

485577

In [38]:
g2p8 = g2p8.dropna() #Removing rows of missing values.

g2p8.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

## Group 2 - Patient 9

In [39]:
g2p9.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89610
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [40]:
len(g2p9) #Size of dataset.

485577

In [41]:
g2p9 = g2p9.dropna() #Removing rows of missing values.

g2p9.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

## Group 2 - Patient 10

In [42]:
g2p10.isna().sum() #Total sum of all missing values for every column.

Composite Element REF        0
Beta_value               89559
Chromosome                   0
Start                        0
End                          0
Gene_Symbol                  0
Gene_Type                    0
Transcript_ID                0
Position_to_TSS              0
CGI_Coordinate               0
Feature_Type                 0
dtype: int64

In [43]:
len(g2p10) #Size of dataset.

485577

In [44]:
g2p10 = g2p10.dropna() #Removing rows of missing values.

g2p10.isna().sum() #Checking that the data is clean and free from NAs.

Composite Element REF    0
Beta_value               0
Chromosome               0
Start                    0
End                      0
Gene_Symbol              0
Gene_Type                0
Transcript_ID            0
Position_to_TSS          0
CGI_Coordinate           0
Feature_Type             0
dtype: int64

# Download All Clean Files:

Lets download all the patient's clean datasets.

In [45]:
g2p1.to_csv('g2p1.csv')
g2np2.to_csv('g2p2.csv')
g2p3.to_csv('g2p3.csv')
g2p4.to_csv('g2p4.csv')
g2p5.to_csv('g2p5.csv')
g2p6.to_csv('g2p6.csv')
g2p7.to_csv('g2p7.csv')
g2p8.to_csv('g2p8.csv')
g2p9.to_csv('g2p9.csv')
g2p10.to_csv('g2p10.csv')

Now that all DNA Methylation data for every patient on group 2 is clean, I will use it to perform exploratory data analysis in part 4.